core/src/test/java/org/jdbi/v3/core/statement/TestPreparedBatch.java

```diff
@@ -275,6 +275,130 @@ public void testBindNull() {
             .execute()).containsExactly(1, 1, 1);
     }
 
+    @Test
+    public void testDefaultBatchChunkSize() {
+        Handle h = h2Extension.getSharedHandle();
+        PreparedBatch batch = h.prepareBatch("insert into something (id, name) values (:id, :name)");
+        assertThat(batch.getBatchChunkSize()).isEqualTo(Integer.MAX_VALUE);
+    }
+
+    @Test
+    public void testSetBatchChunkSize() {
+        Handle h = h2Extension.getSharedHandle();
+        PreparedBatch batch = h.prepareBatch("insert into something (id, name) values (:id, :name)");
+        batch.setBatchChunkSize(100);
+        assertThat(batch.getBatchChunkSize()).isEqualTo(100);
+        batch.setBatchChunkSize(5000);
+        assertThat(batch.getBatchChunkSize()).isEqualTo(5000);
+    }
+
+    @Test
+    public void testInvalidBatchChunkSize() {
+        Handle h = h2Extension.getSharedHandle();
+        PreparedBatch batch = h.prepareBatch("insert into something (id, name) values (:id, :name)");
+        try {
+            batch.setBatchChunkSize(0);
+            fail("Expected IllegalArgumentException for chunk size 0");
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage()).contains("Batch chunk size must be greater than zero");
+        }
+        try {
+            batch.setBatchChunkSize(-1);
+            fail("Expected IllegalArgumentException for negative chunk size");
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage()).contains("Batch chunk size must be greater than zero");
+        }
+    }
+
+    @Test
+    public void testBatchChunkSizeExecution() {
+        Handle h = h2Extension.getSharedHandle();
+        PreparedBatch batch = h.prepareBatch("insert into something (id, name) values (:id, :name)");
```

Comment:
I know we have tons of examples that don't but this should technically be in a try-with-resources block.